lib/Infer/Interpreter.cpp

```diff
@@ -2,6 +2,31 @@
 
 namespace souper {
 
+  bool hasReservedHelper(Inst *I, std::set<Inst *> &Visited) {
+    if (I->K == Inst::ReservedConst || I->K == Inst::ReservedInst)
+      return true;
+    if (I->K == Inst::Var && (I->Name.find(ReservedConstPrefix) != std::string::npos)) {
+      return true;
+    } else if (I->K == Inst::Var && (I->Name.find(ReservedInstPrefix) != std::string::npos)) {
```

Comment:
I think it's better style to avoid the else when the if-part just returns (like you did above, but not right here)